### PR TITLE
removed SSR trpc calls instead SPA for logged in pages

### LINKED ISSUE #6935: [CAL-1006] Experiment: try dashboard as SPA vs SSR
Posted by @KATT

> Heya :wave: come the \[suspense\]([http://app.cal.com|app.cal.com](http://app.cal.com%7Capp.cal.com)> dashboard is server-side rendered? I feel it would feel faster if you did SPA there and you'd ship faster since it would unlock being able to use <[https://trpc.io/docs/suspense](https://trpc.io/docs/suspense)) etc
>
> Below is a almost 3s *blocking* request when toggling between "Availability" and "Booking"

Like, my suggestion would be to get rid of this stuff: [https://github.com/calcom/cal.com/blob/8c150264f498b8e5d5cfedaebc69465c1dfd2aeb/apps/web/pages/availability/index.tsx#L121-L129](https://github.com/calcom/cal.com/blob/8c150264f498b8e5d5cfedaebc69465c1dfd2aeb/apps/web/pages/availability/index.tsx#L121-L129)

IIRC that stuff is there because of i18n-flicker, but I'm sure there's an alternative way of dealing with that

[Suspense (Experimental) | tRPC](https://trpc.io/docs/suspense)

<[https://github.com/calcom/cal.com/blob/8c150264f498b8e5d5cfedaebc69465c1dfd2aeb/apps/web/pages/availability/index.tsx](https://github.com/calcom/cal.com/blob/8c150264f498b8e5d5cfedaebc69465c1dfd2aeb/apps/web/pages/availability/index.tsx) | index.tsx>

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-1006](https://linear.app/calcom/issue/CAL-1006/experiment-try-dashboard-as-spa-vs-ssr)</sub>

### PR DESCRIPTION
**Pull request by PeerRich**
fixes https://github.com/calcom/cal.com/issues/6935

so far so good on localhost

## Why SSR in the first place: 

we used to have i18n flickering with CSR but i cant reproduce them anymore. ideally we dont rely on SSR for logged in pages, since its slower

## what to test

- hard reload some pages
- see if i18n keys are flickering
- change language in settings

### CODE REVIEW

**Comment by AaronPresley** [comment]
The main issue I'm seeing on this is that the page no longer updates the UI strings when changing your preferred language under Settings > General. This felt surprising to me, as I double-checked this workflow as part of [my previous PR](https://github.com/calcom/cal.com/pull/6823).

I checked out `main` and am seeing the same problem, so it's not necessarily related to this PR directly. I see there have been [several commits](https://github.com/calcom/cal.com/commits/main/apps/web/pages/api/trpc/%5Btrpc%5D.ts) relating to header cache since my PR. Might be worth spinning up an issue?

**Review by AaronPresley — APPROVED** [review]
Looking good overall, despite what I mentioned in [this comment](https://github.com/calcom/cal.com/pull/6936#issuecomment-1428882528) which should likely be a new Issue. Otherwise I don't see any flickering.

**Comment by PeerRich** [comment]
>  Might be worth spinning up an issue?

oh yes please! seems like a regression

**Comment by PeerRich** [comment]
hmm checks failing, and tests. not sure why. maybe @zomars knows more

**Review by zomars — CHANGES_REQUESTED** [review]
Blocking to type fix, also this PR is leaving a lot of unused imports

File: apps/web/pages/event-types/index.tsx [hunk 1]
```diff
@@ -606,13 +585,6 @@ const EventTypeListHeading = ({
           </Link>
         )}
       </div>
-      {!profile?.slug && !!teamId && (
-        <button onClick={() => publishTeamMutation.mutate({ teamId })}>
-          <Badge variant="gray" className="mb-1 -ml-2">
-            {t("upgrade")}
-          </Badge>
-        </button>
-      )}
```
Comment (zomars) [inline apps/web/pages/event-types/index.tsx:615]: Is this change intended for this PR¡

**Comment by zomars** [comment]
This breaking Embed tests. cc @hariombalhara to help debugging

File: apps/web/pages/event-types/index.tsx (hunk 1 above)
Comment (PeerRich) [reply apps/web/pages/event-types/index.tsx:615]: no not from me

File: apps/web/pages/event-types/index.tsx [hunk 2]
```diff
@@ -128,17 +125,11 @@ const Item = ({ type, group, readOnly }: { type: EventType; group: EventTypeGrou
           data-testid={"event-type-title-" + type.id}>
           {type.title}
         </span>
-        {group.profile.slug ? (
```
Comment (PeerRich) [inline apps/web/pages/event-types/index.tsx:131]: this is not from me :o

File: apps/web/pages/event-types/index.tsx [hunk 3]
```diff
@@ -128,17 +125,11 @@ const Item = ({ type, group, readOnly }: { type: EventType; group: EventTypeGrou
...
-          </small>
-        ) : (
-          <></>
-        )}
+        <small
+          className="hidden font-normal leading-4 text-gray-600 sm:inline"
+          data-testid={"event-type-slug-" + type.id}>{`/${group.profile.slug}/${type.slug}`}</small>
         {readOnly && (
-          <span className="inline items-center rounded-sm bg-gray-100 px-1.5 py-0.5 text-xs font-medium text-gray-800 ltr:ml-2 ltr:mr-2 rtl:ml-2">
+          <span className="rtl:ml-2inline items-center rounded-sm bg-gray-100 px-1.5 py-0.5 text-xs font-medium text-gray-800 ltr:ml-2 ltr:mr-2">
```
Comment (PeerRich) [inline apps/web/pages/event-types/index.tsx:132]: same here. also typo

```suggestion
          <span className="rtl:ml-2 inline items-center rounded-sm bg-gray-100 px-1.5 py-0.5 text-xs font-medium text-gray-800 ltr:ml-2 ltr:mr-2">
```

File: apps/web/pages/event-types/index.tsx [hunk 4]
```diff
@@ -563,18 +554,6 @@ const EventTypeListHeading = ({
...
-
-  const publishTeamMutation = trpc.viewer.teams.publish.useMutation({
-    onSuccess(data) {
-      router.push(data.url);
-    },
-    onError: (error) => {
-      showToast(error.message, "error");
-    },
-  });
-
```
Comment (PeerRich) [inline apps/web/pages/event-types/index.tsx:577]: also not by me

**Comment by zomars** [comment]
Can you take over this PR @roae

File: apps/web/pages/event-types/index.tsx (hunk 3 above)
Comment (roae) [reply apps/web/pages/event-types/index.tsx:132]: the span tag is already inline, it is redundant

**Review by roae — APPROVED** [review]
Looking good overall, I never saw any flickering, even changing the language in settings

File: apps/web/pages/event-types/index.tsx (hunk 2 above)
Comment (PeerRich) [reply apps/web/pages/event-types/index.tsx:131]: before we merge can we undo this? i dont know where this came from @zomars  @roae

File: apps/web/pages/event-types/index.tsx (hunk 4 above)
Comment (PeerRich) [reply apps/web/pages/event-types/index.tsx:578]: https://github.com/calcom/cal.com/pull/6936/files#r1116335299

File: apps/web/pages/event-types/index.tsx (hunk 1 above)
Comment (PeerRich) [reply apps/web/pages/event-types/index.tsx:616]: https://github.com/calcom/cal.com/pull/6936/files#r1116335299

**Review by zomars — APPROVED** [review]
